### DISCUSSION TITLE
Ontotext open source workbench ?

### DISCUSSION THREAD

**Discussion by barthanssens** [eclipse-rdf4j/rdf4j#3185] [Ideas]
Ontotext has open sourced their workbench https://github.com/Ontotext-AD/graphdb-workbench (Apache License)

Though it is not clear if it also can be used with "plain" RDF4J, would it be an option to support this workbench as an alternative to or replacement of the RDF4J workbench ?

**Comment by hmottestad** [comment]
Got any screenshots?

**Comment by barthanssens** [comment]
Not yet, it appears to be npm based, and of course the npm-version on my cheap ARM-test server is not supported :-/ 

But this (somewhat old) youtube video should give a good impression: https://www.youtube.com/watch?v=eEYEcmaYBBY

**Comment by hmottestad** [comment]
Looks fundamentally similar to our own workbench, but a lot more modern :)

**Comment by barthanssens** [comment]
From what I remember from the GraphDB 8 standard version, one can select and delete triples and it has a nice visualization for ontologies.

Once again, I don't know if it works with "plain" RDF4J, but it's worth the try, maybe @avataar knows more about this.

**Comment by avataar** [comment]
Many things will work fine with plain old RDF4J but some things rely on GraphDB's APIs (e.g. repository managament, user management, most of the visualizations, import). Forking and adapting it to pure RDF4J is certainly possible. We just open sourced what we had, the goal wasn't to make it universal out of the box.

**Comment by abrokenjester** [comment]
We need someone with some Angular experience to take a closer look at this, and give an estimate of what's involved.

**Comment by VladimirAlexiev** [comment]
- see https://graphdb.ontotext.com/documentation/standard/devhub/extend-workbench.html for programmer advice
- @hmottestad screenshots are all over the place in the documentation, eg
https://graphdb.ontotext.com/documentation/standard/loading-data-using-the-workbench.html
https://graphdb.ontotext.com/documentation/standard/loading-data-using-ontorefine.html
https://graphdb.ontotext.com/documentation/standard/querying-data.html

Some specific points that come to mind:
- disabling stuff that's not in rdf4j. In addition to what @avataar wrote: OntoRefine, OntoRefine RDF mapping, Monitoring, most of the Setup operations...
- branding/look
- BTW, is anyone up for internationalization/localization? Some French people want it bilingual (en, fr), and surprisingly I could't find a single SPARQL endpoint/editor that can be translated fully. Not so easy, because it incorporates external libraries (yasgui, google charts, pivot charts) that also need to be internationalized.
